geopandas/tests/test_indexing.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+from geopandas import GeoDataFrame, GeoSeries
+from shapely.geometry import Point
+import numpy as np
+import pandas as pd
+
+
+@pytest.fixture
+def df():
+    return GeoDataFrame(
+        {
+            "geometry": [Point(x, x) for x in range(3)],
+            "geometry2": [Point(x, x) for x in range(3)],
+            "geometry3": [Point(x, x) for x in range(3)],
+            "value": [1, 2, 1],
+            "value_nan": np.nan,
+        }
+    )
+
+
+test_case_column_sets = [
+    ["geometry"],
```

Comment:
I don't think the nan cases are required, but otherwise I saw distinct type(data), data.dtype / successive call combinations for all of these column combinations when I put prints in _geodataframe_constructor_sliced. I skiped ["geometry2", "geometry3"] because it's identical to the single non active geometry column. As an aside I was surprised that the active geometry column made a difference, but it did seem to.